### ISSUE TITLE
Should the plugin template still use undefined passed as a parameter 

### ISSUE THREAD

**Issue by toniedzwiedz** [jquery-boilerplate/jquery-boilerplate#124]
[This file](https://github.com/jquery-boilerplate/jquery-boilerplate/blob/master/src/jquery.boilerplate.js) provides an example of `undefined` being passed as an argument to a function:

```
;( function( $, window, document, undefined ) {
    //omitted for brevity
})($, window, document);
```

The idea here seems to be to make sure `undefined` is really undefined by not passing it into the actual call. However, I believe this is no longer necessary in ES5+ compliant browsers.

These days, it's actually safer not to do it as it introduces a way for a developer to introduce a local variable named `undefined` by accidentally calling the function with too many arguments. [MDN seems to advise against that](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined) and personally, I agree.

Later in the same script, there's a comment saying:

```
    // undefined is used here as the undefined global variable in ECMAScript 3 is
    // mutable (ie. it can be changed by someone else). undefined isn't really being
    // passed in so we can ensure the value of it is truly undefined. In ES5, undefined
    // can no longer be modified.
```

I understand that the fact the template still uses this practice is a conscious decision based on the team's belief in continuing to do it to cater for pre-ES5 browsers. However, I think it should also be mentioned very explicitly that this actually introduces a risk in modern browsers and should only be done if one actually wants to support ES3 browsers.

What do you think?

**Comment by jstnbr** [comment]
I agree with dropping it. The [Javascript Wikipedia](https://en.wikipedia.org/wiki/JavaScript) says "ECMAScript 5 was finally released in December 2009."

That means since December 2009 this technique has been obsolete.
